### PR TITLE
LONDON | GISLAINE_DELLA_BELLA | Module-Structuring-and-Testing-Data | SPRINT 1

### PR DESCRIPTION
**Pull request by Della-Bella**
Self checklist

[X ] I have committed my files one by one, on purpose, and for a reason
[ X] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK
[X ] I have tested my changes
[X ] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/contributing/)
[X ] My changes meet the [requirements](https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/compare/README.md) of this task
Hi volunteer,

Here are my exercises from Sprint 1.

I had to ignore my first branch f and restart again as I was making my changes only in one branch and even following what CJ Juan posted on Slack I couldn't merge the last with the new. So here is going again/exercises Sprint 1.

Thank you for taking the time to review them.
Gislaine

## Changelist

- Recommit all Sprint 1 exercises and create a new PR

### CODE REVIEW

File: Sprint-1/interpret/time-format.js [hunk 1]
```diff
@@ -12,13 +12,25 @@ console.log(result);
 // For the piece of code above, read the code and then answer the following questions
 
 // a) How many variable declarations are there in this program?
+//5
 
 // b) How many function calls are there?
-
+//6
 // c) Using documentation, explain what the expression movieLength % 60 represents
+//divided by 60, it's  convert seconds to a time format
```
Comment (cjyuan) [inline Sprint-1/interpret/time-format.js:20]: This description (and also the descriptions in other comments in this file) are not precise enough. Why don't you ask your new friend ChatGPT for some suggestions.

File: Sprint-1/interpret/to-pounds.js [hunk 2]
```diff
@@ -1,27 +1,36 @@
 const penceString = "399p";
 
 const penceStringWithoutTrailingP = penceString.substring(
-  0,
-  penceString.length - 1
-);
+  0, // extracts a portion of the string here starting from index 0
```
Comment (cjyuan) [inline Sprint-1/interpret/to-pounds.js:4]: Seems to have some typo in this comment.
Some of the other comments are not precise enough.

File: Sprint-1/interpret/to-pounds.js [hunk 3]
```diff
@@ -1,27 +1,36 @@
...
-);
+  0, //and stop with (0) so has to be 3 characters
+  paddedPenceNumberString.length - 2 //gets the characters from the beginning of the string up to 2 characters from the end.
+); // result is stored in paddedPenceNumberString
 
 const pence = paddedPenceNumberString
-  .substring(paddedPenceNumberString.length - 2)
-  .padEnd(2, "0");
+  .substring(paddedPenceNumberString.length - 2) // extracts the last 2 characters from the strin
+  .padEnd(2, "0"); //ensures the string is at least 2 characters long
```
Comment (cjyuan) [inline Sprint-1/interpret/to-pounds.js:16]: Can we expect this program to work as intended even if we deleted `.padEnd(2, "0")` from the code?

**Review by cjyuan — COMMENTED** [review]
I left my comments in the code.

Why not also try questions in the files in folder `explore`?

File: Sprint-1/exercises/count.js [hunk 4]
```diff
@@ -4,3 +4,5 @@ count = count + 1;
 
 // Line 1 is a variable declaration, creating the count variable with an initial value of 0
 // Describe what line 3 is doing, in particular focus on what = is doing
+
+// on line 3 the variable is getting the value of a variable count and adding 1 to it
```
[the remark this answers is not in the document]
Comment (Della-Bella) [reply Sprint-1/exercises/count.js:8]: This is count++;

File: Sprint-1/errors/3.js [hunk 5]
```diff
@@ -1,9 +1,21 @@
...
 // However, the code isn't working
 // Before running the code, make and explain a prediction about why the code won't work
 // Then run the code and see what error it gives.
 // Consider: Why does it give this error? Is this what I predicted? If not, what's different?
 // Then try updating the expression last4Digits is assigned to, in order to get the correct value
+
+// The code isin't work because slice is a method for strings and arrays, but cardNumber is defined as a number.
+// use slice on a number will result in an error.
+
+const cardNumber = "4533787178994213";
```
[the remark this answers is not in the document]
Comment (Della-Bella) [reply Sprint-1/errors/3.js:16]: // New solution without modifying the const cardNumber. I converted it to a String and then used the function Slice because Slice can only be used in a string.

File: Sprint-1/errors/3.js [hunk 6]
```diff
@@ -1,9 +1,21 @@
...
+const last4Digits = cardNumber.slice(-4);
+
+function cardnumber() {
```
[the remark this answers is not in the document]
Comment (Della-Bella) [reply Sprint-1/errors/3.js:19]: done, correct it without modify the const CardNumber

File: Sprint-1/exercises/initials.js [hunk 7]
```diff
@@ -1,6 +1,21 @@
...
+ lled initials that stores the first character of each string.
+This should produce the string "CKJ", but you must not write the characters C, K, or J in the code of your solution.*/
+
 let firstName = "Creola";
 let middleName = "Katherine";
 let lastName = "Johnson";
 
-// Declare a variable called initials that stores the first character of each string.
-// This should produce the string "CKJ", but you must not write the characters C, K, or J in the code of your solution.
+function initials() {
```
[the remark this answers is not in the document]
Comment (Della-Bella) [reply Sprint-1/exercises/initials.js:13]: Done/ New solution assigns it in variables:

let firstInitial = firstName.charAt(0); 
let middleInitial = middleName.charAt(0);
let lastInitial = lastName.charAt(0);

File: Sprint-1/exercises/random.js [hunk 8]
```diff
@@ -1,9 +1,53 @@
...
+
+// Try breaking down the expression and using documentation to explain what it means:
+
+/* Match.floor=  is a JS function which rounds a number down to the nearest integer
+  also it removing the decimals.
+  if number is Positive it will round to the nearest. 4.7 = 4 4.3=4
+  And if the number is negative it will round it to nearest = -4.7 = -5
+
+Match.random= is a js function that generates random numbers 
+between 0 and 1.
```
[the remark this answers is not in the document]
Comment (Della-Bella) [reply Sprint-1/exercises/random.js:19]: The num function will generate a random number between 0 and 100.  Math.random() would include 0 but as we have " **+ minimum**" this will add 1 to the result. so  0 + 1 will never be 0.

File: Sprint-1/exercises/random.js [hunk 9]
```diff
@@ -1,9 +1,53 @@
...
+
+done */
+
+const minimum = 1;
+const maximum = 100;
+
+const num = Math.floor(Math.random() * (maximum - minimum + 1)) + minimum;
+
+console.log(num);
+
```
[the remark this answers is not in the document]
Comment (Della-Bella) [reply Sprint-1/exercises/random.js:41]: To generate a number between -5 and 3 it will be:

const minimum = -5;
const maximum = 3;

const num = Math.floor(Math.random() * (maximum - minimum + 1)) + minimum;

File: Sprint-1/interpret/percentage-change.js [hunk 10]
```diff
@@ -2,21 +2,36 @@ let carPrice = "10,000";
...
+priceAfterOneYear = Number(priceAfterOneYear.replaceAll(",", ""));*/
 
 // c) Identify all the lines that are variable reassignment statements
+//4, 5
 
 // d) Identify all the lines that are variable declarations
+//1,2 7, 8
 
 // e) Describe what the expression Number(carPrice.replaceAll(",","")) is doing - what is the purpose of this expression?
+//it transform a format number price in a number
```
[the remark this answers is not in the document]
Comment (Della-Bella) [reply Sprint-1/interpret/percentage-change.js:34]: This is what chat GTP gave me .. Seen more clear 

 carPrice.replaceAll(",", ""): This part removes all commas (",") from the carPrice string.

Number(...): This attempts to convert the resulting string (without commas) into a numerical value.

The purpose of this expression is to prepare the carPrice string for numerical calculations. 

By removing the commas, it ensures that the string can be correctly interpreted as a number by the Number() function

File: Sprint-1/interpret/time-format.js [hunk 11]
```diff
@@ -12,13 +12,25 @@ console.log(result);
 // For the piece of code above, read the code and then answer the following questions
 
 // a) How many variable declarations are there in this program?
+//5
 
 // b) How many function calls are there?
-
+//6
```
[the remark this answers is not in the document]
Comment (Della-Bella) [reply Sprint-1/interpret/time-format.js:18]: Correct answer:
0;  the code  uses variables (movieLength, remainingSeconds, totalMinutes, remainingMinutes, totalHours, result) to store values.

It performs calculations using arithmetic operators like % (modulo) and / (division).

It uses string interpolation (${...}) to create the formatted output.

File: Sprint-1/interpret/time-format.js (hunk 1 above)
Comment (Della-Bella) [reply Sprint-1/interpret/time-format.js:20]: It is clear now.  is used to show the number of remaining seconds after calculating the total number of minutes in the movie.

"The expression movieLength % 60 in JavaScript represents the modulo operation, which calculates the remainder of the division between movieLength and 60.

In simpler terms:

It divides the movieLength (in seconds) by 60 (the number of seconds in a minute).
The result of this division might have a remainder (e.g., if the movie length is not perfectly divisible by 60).
The % operator specifically extracts this remainder.
Purpose in the Code:

In the given code snippet, movieLength % 60 is used to determine the number of remaining seconds after calculating the total number of minutes in the movie. This is crucial for accurately representing the movie's duration in hours, minutes, and seconds.

Example:

If movieLength is 2784 seconds:

2784 / 60 = 46 with a remainder of 24
movieLength % 60 will result in 24, indicating that there are 24 seconds remaining after 46 full minutes.

File: Sprint-1/interpret/to-pounds.js (hunk 3 above)
Comment (Della-Bella) [reply Sprint-1/interpret/to-pounds.js:16]: I asked cha Gtp and it explain me  that it would work as I expected because, without .padEnd(2, "0"), the pence value would be displayed as a single digit when it's less than 10.

Ex: "399p" would be incorrectly displayed as "£3.9" instead of the correct "£3.99".

File: Sprint-1/interpret/to-pounds.js (hunk 2 above)
Comment (Della-Bella) [reply Sprint-1/interpret/to-pounds.js]: I understand clearly now asking to chat GTP line by line. 
Revised code and explanation in VS

**Comment by Della-Bella** [comment]
Hi cjyuan,
I have revised all the exercises, and they are now clearer to me than they were when I did it. 
thank you

**Comment by Della-Bella** [comment]
Sorry, what should I do here with this conflict?

**Comment by Della-Bella** [comment]
ok thank you 
have a nice day ;)
